.github/workflows/promote.yml

```diff
@@ -0,0 +1,105 @@
+name: promote
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        description: full npm version number, like 7.150.2
+        required: true
+      tag:
+        type: string
+        required: false
+        description: tag to use for docker/npm, like latest or latest-rc
+        default: latest
+      channel:
+        type: string
+        required: true
+        default: stable
+        description: channel to promote the matching tarball to (example stable, stable-rc )
+      useCTC:
+        type: boolean
+        required: false
+        description: this promotion should only run if no release moratorium is in place
+
+jobs:
+  ctc-open:
+    if: inputs.useCTC
+    uses: salesforcecli/github-workflows/.github/workflows/ctcOpen.yml@main
+    secrets: inherit
```

Comment:
If `useCTC` is set to `false` this step will skip... How do the jobs below that have `needs: [ctc-open]` respond to this? Is a "skip" a failure that would prevent a `needs` check?

Reply:
Ok, i think i answered my own question. Using `if` will mark a step as "skipped" ([docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif)). And `needs` requires a step to finish "successfully" ([docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds)).

Reply:
yeah, it works that way: https://github.com/actions/runner/issues/491